catalog/files/api/src/main/java/org/projectnessie/catalog/files/api/StorageFailureException.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.catalog.files.api;
+
+import org.projectnessie.storage.uri.StorageUri;
+
+public class StorageFailureException extends NonRetryableException {
```

Comment:
I think the "hard"/unconditional association with "non-retryable" is not good.
We need a way to distinguish "hard/final failures" from "retryable" ones, so that `org.projectnessie.catalog.service.impl.Util#throwableAsErrorTaskState` (via e.g. `org.projectnessie.catalog.service.impl.EntitySnapshotTaskBehavior#asErrorTaskState`) can return the right state (retryable vs final).

Retryable examples:
* Throttled requests
* Technical network errors ("no route to host", "read timeout")
* Other unknown (IO)Exceptions
* Forbidden (users may "fix" credentials or ACLs)

Non-retryable ones:
* Not found

We might eventually need a third TaskState - to handle "forbidden"/"access denied" better, to handle the case when the object store's ACL's just wrong: a state that's immediately reported as an error, but retried later.

Reply:
> We might eventually need a third TaskState - to handle "forbidden"/"access denied" better, to handle the case when the object store's ACL's just wrong: a state that's immediately reported as an error, but retried later.

Isn't that what `TaskStatus.ERROR_RETRY` already does?

Reply:
To mitigate potentially unmapped/unhandled error states, WDYT about making all current `FAILURE` states `ERROR_RETRY` ones but with a somewhat higher retry interval?

Reply:
> Non-retryable ones:    Not found

Actually, this feels also retryable. Sure, in a perfect world, "not found" should be a "final" state.
But it could also be an ephemeral case, for example when users messed up the object store locations - in theory though, but possible. Or if someone manually migrates files.

Reply:
Yes, that's pretty much where I'm moving (in my local ATM). Each caller should timeout at some point and concurrent caller would get the same result within some time window. If another caller, touches the same task after a certain time period, a re-try will happen regardless of the root cause. We can tune with the first and second timeout separately for each "cause".

Reply:
Re: "not found" : It may be in reality represent a permission error (e.g. in GH), so we cannot really rely on the object being really missing.